File: backend/app/models/isp_package.py
from datetime import datetime
from typing import Optional
from app.schemas.isp_package import IspManagerPackageType
import strawberry

@strawberry.type
class DBISPPackage:
    _id: str
    name: str
    description: str
    price: float
    organizationId: str  # Just the ID in the database model
    downloadSpeed: float
    uploadSpeed: float
    burstDownload: Optional[float]
    burstUpload: Optional[float]
    thresholdDownload: Optional[float]
    thresholdUpload: Optional[float]
    burstTime: Optional[int]
    serviceType: IspManagerPackageType
    addressPool: Optional[str]
    # Session management
    sessionTimeout: Optional[int]
    idleTimeout: Optional[int]
    # QoS and VLAN
    priority: Optional[int]
    vlanId: Optional[int]
    createdAt: datetime
    updatedAt: datetime

File: frontend/types/isp_package.ts
export type ISPPackage = {
  id: string;
  name: string;
  description?: string;
  price: number;
  organization: {
    id: string;
    name: string;
  };
  downloadSpeed: number;
  uploadSpeed: number;
  burstDownload: number;
  burstUpload: number;
  thresholdDownload: number;
  thresholdUpload: number;
  burstTime: number;
  serviceType: 'PPPOE' | 'HOTSPOT' | 'STATIC' | 'DHCP';
  status: 'ACTIVE' | 'INACTIVE' | 'DRAFT';
  addressPool: string;
  createdAt: string;
  updatedAt: string;
};

export interface ISPPackagesResponse {
  packages: {
    success: boolean;
    message: string;
    packages: ISPPackage[];
  };
}

export interface ISPPackageResponse {
  package: {
    success: boolean;
    message: string;
    package: ISPPackage;
  };
}

export type CreateISPPackageInput = {
  name: string;
  description?: string;
  price: number;
  organizationId: string;
  downloadSpeed: number;
  uploadSpeed: number;
  burstDownload: number;
  burstUpload: number;
  thresholdDownload: number;
  thresholdUpload: number;
  burstTime: number;
  serviceType: string;
  addressPool: string;
};

export type UpdateISPPackageInput = Partial<CreateISPPackageInput>;

